File: framework/languages/java/e2e/RegistrationCodec/Client/src/main/java/systems/zlink/e2e/registrationcodec/client/Scenarios/RcB4CodecCoexistenceScenario.java
package systems.zlink.e2e.registrationcodec.client.Scenarios;

import systems.zlink.e2e.registrationcodec.client.Support.ScenarioAssert;
import systems.zlink.e2e.registrationcodec.client.Support.ScenarioContext;
import systems.zlink.e2e.registrationcodec.shared.Contracts;

public final class RcB4CodecCoexistenceScenario {
    private RcB4CodecCoexistenceScenario() {
    }

    public static void run(ScenarioContext context) {
        context.server().post("/codec/roundtrip").submit(Contracts.CodecScenarioRes.class).toCompletableFuture().join().body();
        ScenarioAssert.waitForEvidence(context.evidence(), "ContentType", "JsonEcho", "application/json");
        ScenarioAssert.waitForEvidence(
            context.evidence(),
            "ContentType",
            "ProtobufEcho",
            "application/x-protobuf");
        ScenarioAssert.waitForEvidence(
            context.evidence(),
            "ContentType",
            "MsgpackEcho",
            "application/x-msgpack");
        System.out.println("scenario RC-B4 passed");
    }
}

File: framework/languages/java/e2e-kotlin/RegistrationCodec/Client/src/main/kotlin/systems/zlink/e2e/kotlin/registrationcodec/client/Scenarios/RcB4CodecCoexistenceScenario.kt
package systems.zlink.e2e.kotlin.registrationcodec.client.scenarios

import systems.zlink.e2e.kotlin.registrationcodec.CodecRoundtripRes
import systems.zlink.e2e.kotlin.registrationcodec.client.support.ScenarioAssert
import systems.zlink.e2e.kotlin.registrationcodec.client.support.ScenarioHttpClient

class RcB4CodecCoexistenceScenario(
    private val server: ScenarioHttpClient,
    private val assert: ScenarioAssert,
) {
    fun run() {
        val reply = server.post<CodecRoundtripRes>("/codec/roundtrip")
        assert.that(reply.jsonValue == "echo:json-request", "RC-B4 JSON reply mismatch")
        assert.that(reply.protobufValue == "echo:protobuf-request", "RC-B4 Protobuf reply mismatch")
        assert.that(reply.messagePackValue == "echo:msgpack-request", "RC-B4 MessagePack reply mismatch")
        println("scenario RC-B4 passed")
    }
}
